dev/com.ibm.ws.install/src/com/ibm/ws/install/maven/MavenDirectoryInstall.java

```diff
@@ -0,0 +1,276 @@
+			logger.log(Level.FINE, "The list of resolved features is empty.");
+			String exceptionMessage = (String) map.get("action.error.message");
+			if (exceptionMessage == null) {
+				logger.log(Level.INFO, Messages.INSTALL_KERNEL_MESSAGES.getLogMessage("ALREADY_INSTALLED",
+						map.get("features.to.resolve")));
+				return;
+			} else if (exceptionMessage.contains("CWWKF1250I")) {
+				logger.log(Level.INFO, exceptionMessage);
+				return;
+			} else {
+				throw new InstallException(exceptionMessage);
+			}
+		}
+
+		Collection<File> artifacts = new ArrayList<File>();
+		artifacts.addAll(findEsas(resolvedFeatures, fromDir));
+
+
+		Collection<String> actionReturnResult = new ArrayList<String>();
+		Collection<String> currentReturnResult;
+
+		logger.log(Level.INFO, Messages.INSTALL_KERNEL_MESSAGES.getLogMessage("STATE_STARTING_INSTALL"));
+		for (File esaFile : artifacts) {
+
+			logger.log(Level.FINE, Messages.INSTALL_KERNEL_MESSAGES.getLogMessage("STATE_INSTALLING",
+					extractFeature(esaFile.getName())));
+			map.put("license.accept", true);
+			map.put("action.install", esaFile);
+			if (toExtension != null) {
+				map.put("to.extension", toExtension);
+				logger.log(Level.FINE, "Installing to extension: " + toExtension);
+			}
+
+			Integer ac = (Integer) map.get("action.result");
+			logger.log(Level.FINE, "action.result:" + ac);
+			logger.log(Level.FINE, "action.error.message:" + map.get("action.error.message"));
+
+
+			if (map.get("action.error.message") != null) {
+				// error with installation
+				logger.log(Level.FINE, "action.exception.stacktrace: " + map.get("action.error.stacktrace"));
+				String exceptionMessage = (String) map.get("action.error.message");
+				throw new InstallException(exceptionMessage);
+			} else if ((currentReturnResult = (Collection<String>) map.get("action.install.result")) != null) {
+				// installation was successful
+				if (!currentReturnResult.isEmpty()) {
+					actionReturnResult.addAll((Collection<String>) map.get("action.install.result"));
+					logger.log(Level.INFO,
+							(Messages.INSTALL_KERNEL_MESSAGES
+									.getLogMessage("LOG_INSTALLED_FEATURE", String.join(", ", currentReturnResult))
+									.replace("CWWKF1304I: ", ""))); // TODO: come up with new message for successfully
+																	// installed feature
+
+				}
+			}
+		}
+		logger.log(Level.INFO, Messages.INSTALL_KERNEL_MESSAGES.getLogMessage("TOOL_INSTALLATION_COMPLETED"));
+	}
+
+
+	private Collection<File> findEsas(Collection<String> resolvedFeatures, File rootDir) throws InstallException {
+		Collection<File> foundEsas = new HashSet<>();
+		Collection<String> featuresClone = new ArrayList<>(resolvedFeatures);
+		
+		for (String feature : resolvedFeatures) {
+			logger.log(Level.FINE, "Processing feature: " + feature);
+			File groupDir;
+			String mavenCoordinate = feature.split(":")[0];
+			String featureName = feature.split(":")[1];
+
+			if (mavenCoordinate.equals(OPEN_LIBERTY_GROUP_ID)) {
```

Comment:
According to last review comment, we have to search all esa from given root dir.